server/clip_server/model/model.py

```diff
@@ -465,170 +210,151 @@ class CLIPTextCfg:
-        # pytorch uses additive attention mask; fill with -inf
-        mask = torch.empty(self.context_length, self.context_length)
-        mask.fill_(float("-inf"))
-        mask.triu_(1)  # zero out the lower diagonal
-        return mask
-
-    @property
-    def dtype(self):
-        return self.visual.conv1.weight.dtype
-
-    def lock_image_tower(self, unlocked_groups=0, freeze_bn_stats=False):
-        # lock image tower as per LiT - https://arxiv.org/abs/2111.07991
-        self.visual.lock(
-            unlocked_groups=unlocked_groups, freeze_bn_stats=freeze_bn_stats
-        )
-
-    @torch.jit.ignore
-    def set_grad_checkpointing(self, enable=True):
-        self.visual.set_grad_checkpointing(enable)
-        self.transformer.grad_checkpointing = enable
-
-    def encode_image(self, image):
-        return self.visual(image.type(self.dtype))
-
-    def encode_text(self, text):
-        x = self.token_embedding(text).type(self.dtype)  # [batch_size, n_ctx, d_model]
-
-        x = x + self.positional_embedding.type(self.dtype)
-        x = x.permute(1, 0, 2)  # NLD -> LND
-        x = self.transformer(x, attn_mask=self.attn_mask)
-        x = x.permute(1, 0, 2)  # LND -> NLD
-        x = self.ln_final(x).type(self.dtype)
-
-        # x.shape = [batch_size, n_ctx, transformer.width]
-        # take features from the eot embedding (eot_token is the highest number in each sequence)
-        x = x[torch.arange(x.shape[0]), text.argmax(dim=-1)] @ self.text_projection
-
-        return x
-
-    def forward(self, image, text):
-        if image is None:
-            return self.encode_text(text)
-        elif text is None:
-            return self.encode_image(image)
-        image_features = self.encode_image(image)
-        image_features = F.normalize(image_features, dim=-1)
-
-        text_features = self.encode_text(text)
-        text_features = F.normalize(text_features, dim=-1)
-
-        return image_features, text_features, self.logit_scale.exp()
 
+class CustomTextCLIP(_CustomTextCLIP):
+    def __init__(
+        self,
+        embed_dim: int,
+        vision_cfg: CLIPVisionCfg,
+        text_cfg: CLIPTextCfg,
+        quick_gelu: bool = False,
+        dtype: Optional[torch.dtype] = torch.float32,
+    ):
+        super().__init__(embed_dim, vision_cfg, text_cfg, quick_gelu, dtype)
+        self.visual = _build_vision_tower(embed_dim, vision_cfg, quick_gelu, dtype)
+        self.text = _build_text_tower(embed_dim, text_cfg, quick_gelu, dtype)
 
-def convert_weights_to_fp16(model: nn.Module):
-    """Convert applicable model parameters to fp16"""
 
-    def _convert_weights_to_fp16(l):
+class CLIP(_CLIP):
+    def __init__(
+        self,
+        embed_dim: int,
+        vision_cfg: CLIPVisionCfg,
+        text_cfg: CLIPTextCfg,
+        quick_gelu: bool = False,
+        dtype: Optional[torch.dtype] = torch.float32,
+    ):
+        super().__init__(embed_dim, vision_cfg, text_cfg, quick_gelu, dtype)
```

Comment:
It is not necessary to have `super().__init__` here. All of the initializations are included in below.